caas/kubernetes/provider/bootstrap.go

```diff
@@ -403,17 +403,7 @@ func (c *controllerStack) Deploy() (err error) {
 		return errors.Annotate(err, "creating namespace for controller stack")
 	}
 
-	// Check context manually for cancellation between each step (not ideal,
-	// but it avoids wiring context absolutely everywhere).
```

Comment:
Maybe we should be passing the context in / checking in each step as suggested here?

Reply:
I'm fine with doing that, but that's beyond the scope of this PR. I think the solution is to just bite the bullet and pass context to every function. That's not a small task, but one that we could do in a piecemeal fashion. For example, if you're editing a function that doesn't have a first argument that isn't context, add one, with one exception; pure functions don't need this requirement.